### DISCUSSION TITLE
Looking for documentations on how the extension manages data

### DISCUSSION THREAD

**Discussion by fRanK45jPm** [rfennell/AzurePipelines#1584] [General]
Hello, I am looking for any documentations on how the extension is managing Azure DevOps data?

Thank you
John

**Comment by rfennell** [comment]
You have not state which of my extensions you are referencing, but all of my extension follow the same principles

All tasks contain an identical `privacy.md` file (an example is this one on the [ArtifactDescription task](https://github.com/rfennell/AzurePipelines/blob/main/Extensions/ArtifactDescription/privacy.md)), this file states

>We are committed to protecting the security of your information. No data is stored by the extension.
>
>**Data for Quality and Reliability Improvements**
>We do not collect any information to improve the quality and reliability of the extension, nor do we collect telemetry about usage, performance and errors.
>
>**Collection and Use of Your Personal Information**
>We do not collect any information that identifies you as an individual, except the information Microsoft provides to us for commercial purposes
>
>**Disclosure to Third Parties**
Except as described in this statement, information you provide will not be transferred to third parties, Microsoft excluded, without your consent. We may access or disclose information about you, including the content of your communications, in order to: (a) comply with the law or respond to lawful requests or legal process; (b) protect the rights or property of Microsoft or our customers, including the enforcement of our agreements or policies governing your use of the services; or (c) act on a good faith belief that such access or disclosure is necessary to protect the personal safety of our employees, customers, or the public. We may also disclose personal information as part of a corporate transaction such as a merger or sale of assets.

The source for all my tasks can be found in this GitHub project https://github.com/rfennell/AzurePipelines/
